src/commands/job/spec/job.spec.server.ts

```diff
@@ -12,106 +9,171 @@ import {
   Target,
   Logger,
   LogLevel,
-  generateTimestamp
+  generateTimestamp,
+  AuthConfig
 } from '@sasjs/utils'
 import {
+  getAuthConfig,
   removeFromGlobalConfig,
   saveToGlobalConfig
 } from '../../../utils/config'
 import { Command } from '../../../utils/command'
 import {
   createTestApp,
   mockProcessExit,
-  removeTestApp
+  removeTestApp,
+  removeTestServerFolder
 } from '../../../utils/test'
-import { getConstants } from '../../../constants'
 import SASjs, { NoSessionStateError } from '@sasjs/adapter/node'
+import { setConstants } from '../../../utils'
+import { execute } from '../execute'
+import { prefixAppLoc } from '../../../utils/prefixAppLoc'
+import { build } from '../../build/build'
+import { deploy } from '../../deploy/deploy'
 
 describe('sasjs job execute', () => {
   let target: Target
+  let sasjs: SASjs
+  let authConfig: AuthConfig
 
   beforeAll(async () => {
     target = await createGlobalTarget()
     await createTestApp(__dirname, target.name)
     await copyJobsAndServices(target.name)
-    await compileBuildDeployServices(new Command(`cbd -t ${target.name} -f`))
+    await build(target)
+    await deploy(target, false)
 
     await saveToGlobalConfig(
       new Target({
         ...target.toJson()
       })
     )
 
+    sasjs = new SASjs({
+      serverUrl: target.serverUrl,
+      allowInsecureRequests: target.allowInsecureRequests,
+      appLoc: target.appLoc,
+      serverType: target.serverType,
+      debug: true
+    })
+    authConfig = await getAuthConfig(target)
     process.logger = new Logger(LogLevel.Off)
   })
 
   afterAll(async () => {
-    await folder(
-      new Command(
-        `folder delete /Public/app/cli-tests/${target.name} -t ${target.name}`
-      )
-    )
+    await removeTestServerFolder(`/Public/app/cli-tests/${target.name}`, target)
     await removeTestApp(__dirname, target.name)
     await removeFromGlobalConfig(target.name)
   })
 
   it('should submit a job for execution', async () => {
-    const command = new Command(
-      `job execute /Public/app/cli-tests/${target.name}/services/testJob/job -t ${target.name}`
-    )
-
-    await expect(processJob(command)).toResolve()
+    await expect(
```

Comment:
would be nice to put it in function to reduce code duplication

Reply:
refactored specs, fixed.